docs/source/ffi/ffi.rst

```diff
@@ -88,17 +88,19 @@ returns a primitive IO action:
 
 Internally, ``PrimIO Int`` is a function which takes the current (linear)
 state of the world, and returns an ``Int`` with an updated state of the world.
-We can convert this into an ``IO`` action using ``primIO``:
+In general, ``IO`` operations in an Idris program are defined as instances
+of the ``HasIO`` interface (or ``MonadIO``). We can convert a primitive
+operation to one usable in ``HasIO`` using ``primIO``:
 
 .. code-block:: idris
 
-    primIO : PrimIO a -> IO a
+    primIO : HasIO a => PrimIO a -> io a
```

Comment:
Should say `HasIO io` iiuc

Reply:
oh my. How does this even work?

Reply:
Oh right, it's the documentation :). I'll fix it.